src/chainlink/VRFSubscription.ts

```diff
@@ -0,0 +1,106 @@
+import { writeContract } from "../core/helpers/util";
+
+
+export class VRFSubscription {
+    public id: string = "";
+    public balance: string = "";
+    public requestCount: string = "";
+    public owner: string = "";
+    public consumers: string[] = [];
+    public coordinatorAddress: string = "";
+    public chainlinkTokenContractAddress: string = "";
+
+    constructor() { }
+
+    public async fund(amount: string) {
+        await writeContract(
```

Comment:
Set the return value of the `fund()` function to be the same as the return value of the `writeContract()` function

```
public async fund(amount: string): Promise<ContractCallAction> {
    return await writeContract(...
}
```

Do the same for all the other state changing functions on VRFCoordinator & VRFSubscription objects.